internal/cli/runner_install.go

```diff
@@ -0,0 +1,259 @@
+  attempt to install a runner for the server configured in the current Waypoint 
+  context.
+
+` + c.Flags().Help())
+}
+
+func (c *RunnerInstallCommand) Run(args []string) int {
+	ctx := c.Ctx
+	log := c.Log.Named("install")
+	defer c.Close()
+
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoLocalServer(), // no auth in local mode
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+
+	p, ok := runnerinstall.Platforms[strings.ToLower(c.platform)]
+	if !ok {
+		if c.platform == "" {
+			c.ui.Output(
+				"The -platform flag is required.",
+				terminal.WithErrorStyle(),
+			)
+			return 1
+		}
+
+		c.ui.Output(
+			"Error installing server into %q: unsupported platform",
+			c.platform,
+			terminal.WithErrorStyle(),
+		)
+
+		return 1
+	}
+
+	sg := c.ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Connecting to: %s", c.serverUrl)
+	defer func() { s.Abort() }()
+
+	if c.adopt && c.serverCookie == "" {
+		c.ui.Output(
+			"Server cookie must be supplied for adoption.",
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	client := c.project.Client()
+	s.Status(terminal.StatusOK)
+	s.Done()
+
+	// We generate the ID if the user doesn't provide one
+	// This ID is used later to adopt the runner
+	var id string
+	var err error
+	if c.id == "" {
+		id, err = server.Id()
+		if err != nil {
+			c.ui.Output("Error generating runner ID: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+	} else {
+		id = c.id
+	}
+
+	s = sg.Add("Installing runner...")
+	err = p.Install(ctx, &runnerinstall.InstallOpts{
+		Log:        log,
+		UI:         c.ui,
+		Cookie:     c.serverCookie,
+		ServerAddr: c.serverUrl,
+		AdvertiseClient: &serverconfig.Client{
+			Address:       c.serverUrl,
+			Tls:           c.serverTls,
+			TlsSkipVerify: c.serverTlsSkipVerify,
+			RequireAuth:   c.serverRequireAuth,
+		},
+		Id: id,
+	})
+	if err != nil {
+		c.ui.Output("Error installing runner: %s", clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+	s.Update("Runner %s installed successfully", id)
+	s.Status(terminal.StatusOK)
+	s.Done()
+
+	if c.adopt {
+		err = installutil.AdoptRunner(ctx, c.ui, client, id)
+		if err != nil {
+			c.ui.Output("Error adopting runner: %s", clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+			return 1
+		}
+
+		s = sg.Add("Creating runner profile and targeting runner %s", strings.ToUpper(id))
+		// Creating a new runner profile for the newly adopted runner
+		runnerProfile, err := client.UpsertOnDemandRunnerConfig(ctx, &pb.UpsertOnDemandRunnerConfigRequest{
+			Config: &pb.OnDemandRunnerConfig{
+				Name: c.platform + "-" + strings.ToUpper(id),
+				TargetRunner: &pb.Ref_Runner{
+					Target: &pb.Ref_Runner_Id{
+						Id: &pb.Ref_RunnerId{
+							Id: strings.ToUpper(id),
+						},
+					},
+				},
+				OciUrl:               "hashicorp/waypoint-odr:latest",
```

Comment:
I think this should be configurable? I believe our server install includes flags for these already.